Framework/Nexus/test/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ if(CXXTEST_FOUND)
 
   cxxtest_add_test(NexusTest ${TEST_FILES})
 
-  target_link_libraries(NexusTest PRIVATE Mantid::Nexus gmock)
+  target_link_libraries(NexusTest PRIVATE Mantid::Nexus gmock ${HDF5_LIBRARIES})
```

Comment:
I'm surprised it needs to explicitly link HDF5 in the test since Mantid::Nexus already links to it.